src/gt4py/next/program_processors/runners/dace_fieldview/gtir_to_tasklet.py

```diff
@@ -173,10 +213,183 @@ def _visit_deref(self, node: gtir.FunCall) -> MemletExpr | ValueExpr:
+        )
+
+        neighbors_temp, _ = self.sdfg.add_temp_transient(
+            (offset_provider.max_neighbors,), field_desc.dtype
+        )
+        neighbors_node = self.state.add_access(neighbors_temp)
+
+        neighbors_field_type = dace_fieldview_util.get_neighbors_field_type(
+            offset, field_desc.dtype
+        )
+        neighbor_idx = dace_fieldview_util.get_map_variable(neighbors_field_type.dims[0])
+        me, mx = self._add_map(
+            f"{offset}_neighbors",
+            {
+                neighbor_idx: f"0:{offset_provider.max_neighbors}",
+            },
+        )
+        if offset_provider.has_skip_values:
+            assert self.reduce_identity is not None
+            assert self.reduce_identity.dtype == field_desc.dtype
+            skip_value_code = f" if __index != {gtx_common._DEFAULT_SKIP_VALUE} else {self.reduce_identity.dtype}({self.reduce_identity.value})"
+        else:
+            skip_value_code = ""
+        tasklet_node = self._add_tasklet(
+            "gather_neighbors",
+            {"__field", "__index"},
+            {"__val"},
+            "__val = __field[__index]" + skip_value_code,
+        )
+        self.state.add_memlet_path(
+            field_slice_node,
+            me,
+            tasklet_node,
+            dst_conn="__field",
+            memlet=dace.Memlet.from_array(field_slice_view, field_slice_desc),
+        )
+        self.state.add_memlet_path(
+            connectivity_slice_node,
+            me,
+            tasklet_node,
+            dst_conn="__index",
+            memlet=dace.Memlet(data=connectivity_slice_view, subset=neighbor_idx),
+        )
+        self.state.add_memlet_path(
+            tasklet_node,
+            mx,
+            neighbors_node,
+            src_conn="__val",
+            memlet=dace.Memlet(data=neighbors_temp, subset=neighbor_idx),
+        )
+        return ValueExpr(neighbors_node, neighbors_field_type)
+
+    def _visit_reduce(self, node: gtir.FunCall) -> ValueExpr:
+        op_name, reduce_init, reduce_identity = get_reduce_params(node)
+        dtype = reduce_identity.dtype
+
+        # we store the value of reduce identity in the visitor context while visiting the input to reduction
+        # this value will be returned by the neighbors builtin function for skip values
+        prev_reduce_identity = self.reduce_identity
+        self.reduce_identity = reduce_identity
+        assert len(node.args) == 1
+        input_expr = self.visit(node.args[0])
+        assert isinstance(input_expr, MemletExpr | ValueExpr)
+        self.reduce_identity = prev_reduce_identity
+        input_desc = input_expr.node.desc(self.sdfg)
+        assert isinstance(input_desc, dace.data.Array)
+
+        if len(input_desc.shape) > 1:
+            assert isinstance(input_expr, MemletExpr)
+            ndims = len(input_desc.shape) - 1
+            assert set(input_expr.subset.size()[0:ndims]) == {1}
+            reduce_axes = [ndims]
```

Comment:
Just out of curiosity this means that the axis you reduce is always the last axis?